src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetSetOrchestrationServiceStateMethod.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            ExecuteClientAction(() =>
+            {
+                if (ShouldProcess(this.VMScaleSetName, VerbsCommon.Set))
+                {
+                    string resourceGroupName;
+                    string vmScaleSetName;
+                    switch (this.ParameterSetName)
+                    {
+                        case "ResourceIdParameter":
+                            resourceGroupName = GetResourceGroupName(this.ResourceId);
+                            vmScaleSetName = GetResourceName(this.ResourceId, "Microsoft.Compute/virtualMachineScaleSets");
+                            break;
+                        case "ObjectParameter":
+                            resourceGroupName = GetResourceGroupName(this.InputObject.Id);
+                            vmScaleSetName = GetResourceName(this.InputObject.Id, "Microsoft.Compute/virtualMachineScaleSets");
+                            break;
+                        default:
+                            resourceGroupName = this.ResourceGroupName;
+                            vmScaleSetName = this.VMScaleSetName;
+                            break;
+                    }
+                    string action = this.Action;
+
+                    var result = VirtualMachineScaleSetsClient.SetOrchestrationServiceStateWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, action).GetAwaiter().GetResult();
+                    PSOperationStatusResponse output = new PSOperationStatusResponse
+                    {
+                        StartTime = this.StartTime,
+                        EndTime = DateTime.Now
+                    };
+
+                    if (result != null && result.Request != null && result.Request.RequestUri != null)
+                    {
+                        output.Name = GetOperationIdFromUrlString(result.Request.RequestUri.ToString());
+                    }
+
+                    WriteObject(output);
+                }
+            });
+        }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/virtualMachineScaleSets", "ResourceGroupName")]
+        [Alias("Name")]
+        public string VMScaleSetName { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateSet("AutomaticRepairs")]
+        public string ServiceName { get; set; }
```

Comment:
If the ServerName is to add more values in the further, argument completer will be a better choice than validate set